crates/plugins/src/manager.rs

```diff
@@ -194,6 +194,48 @@ impl PluginManager {
         };
         Ok(plugin_manifest)
     }
+
+    pub async fn update_lock(&self) -> PluginManagerUpdateLock {
+        let lock = self.update_lock_impl().await;
+        PluginManagerUpdateLock { lock }
+    }
+
+    async fn update_lock_impl(&self) -> anyhow::Result<fd_lock::RwLock<tokio::fs::File>> {
+        let plugins_dir = self.store().get_plugins_directory();
+        tokio::fs::create_dir_all(plugins_dir).await?;
+        let file = tokio::fs::File::create(plugins_dir.join(".updatelock")).await?;
+        let locker = fd_lock::RwLock::new(file);
+        Ok(locker)
+    }
+}
+
+pub struct PluginManagerUpdateLock {
+    lock: anyhow::Result<fd_lock::RwLock<tokio::fs::File>>,
+}
+
+impl PluginManagerUpdateLock {
+    pub fn lock_updates(&mut self) -> PluginManagerUpdateGuard<'_> {
+        if let Ok(lock) = self.lock.as_mut() {
+            let guard = Some(lock.try_write());
+            PluginManagerUpdateGuard { guard }
+        } else {
+            PluginManagerUpdateGuard { guard: None }
+        }
+    }
+}
+
+pub struct PluginManagerUpdateGuard<'lock> {
+    guard: Option<std::io::Result<fd_lock::RwLockWriteGuard<'lock, tokio::fs::File>>>,
```

Comment:
Same as above. Why do we keep the `io::Result`?

Reply:
`try_write` returns `io::Result<RwLockWriteGuard<'_, T>>` and the result appears to be what we want to query to check for the `WouldBlock` error

Reply:
Indeed.  As above, we could resolve "does it mean denied" at creation time, and create a custom enum a la `Denied / DidNotLock / Locked(guard)`.  I was worried I was already being profligate with new types but maybe that would be worth it.